DataSetExplorer/DataSetBuilder/DataSetInstanceService.cs

```diff
@@ -0,0 +1,36 @@
+﻿using DataSetExplorer.Controllers.DataSetInstance.DTOs;
+using DataSetExplorer.DataSetBuilder.Model;
+using DataSetExplorer.DataSetBuilder.Model.Exceptions;
+using DataSetExplorer.DataSetBuilder.Model.Repository;
+using FluentResults;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace DataSetExplorer.DataSetBuilder
+{
+    public class DataSetInstanceService : IDataSetInstanceService
+    {
+        private readonly IDataSetInstanceRepository _dataSetInstanceRepository;
+
+        public DataSetInstanceService(IDataSetInstanceRepository dataSetInstanceRepository)
+        {
+            _dataSetInstanceRepository = dataSetInstanceRepository;
+        }
+
+        public Result<string> AddDataSetAnnotation(DataSetAnnotationDTO annotation, int annotatorId)
```

Comment:
Service classes should not work with DTOs but domain objects. Use AutoMapper https://docs.automapper.org/en/stable/Getting-started.html to transform the DTO into a domain object in the controller.

For examples, see e.g. https://github.com/Clean-CaDET/platform/blob/master/SmartTutor/Controllers/Progress/SubmissionController.cs#L34

Fix this across the board.